neqo-transport/src/server.rs

```diff
@@ -573,6 +583,16 @@ impl Server {
                 packet.wire_version(),
                 self.conn_params.get_versions().all(),
             );
+
+            crate::qlog::server_version_information_failed(
+                &mut self.create_qlog_trace(&AttemptKey {
+                    remote_address: dgram.source(),
+                    odcid: packet.dcid().into(),
+                }),
+                self.conn_params.get_versions().all(),
+                packet.wire_version(),
```

Comment:
`packet.wire_version()` expects `packet.version` to be `Some`.

https://github.com/mozilla/neqo/blob/abf2636153d5c88feee5faf5663facc79defc858/neqo-transport/src/packet/mod.rs#L694-L697

Based on `PublicPacket::decode` this is valid for `PacketType::OtherVersion` and `PacketType::Initial`.

https://github.com/mozilla/neqo/blob/abf2636153d5c88feee5faf5663facc79defc858/neqo-transport/src/packet/mod.rs#L553-L642

Suggestions for a way to obtain the version without this implicit assumption? I am afraid it might turn into a footgun with future refactorings.

Reply:
I'm not seeing a problem here.  We will have `Some` version in this case, so the assertion should pass.  The only case where we have a `None` version is `VersionNegotiation`, which is inherently not a version.  The effect is only felt if people get a wire version from `VersionNegotiation`, which they shouldn't do.  And the true effect, if assertions are disabled, is that you get version 0, which is correct.